bug: metadata might not always be available

**Describe the bug**
During TestHandlePeerTopicEvent() execution nil pointer error occurred. Problem is with newly added Waku metadata check in  handlerPeerTopicEvent(). This check pm.metadata.RelayShard() assumes metadata are always set. 

**To Reproduce**
1) Checkout https://github.com/waku-org/go-waku/pull/1035/commits/f8a67f83eab33dfa09ec308536a496381d02756c from https://github.com/waku-org/go-waku/pull/1035
2) Run go test

**Expected behavior**
Please add nil metadata situation handling or ensure metadata are never nil with example how to instantiate when no cluster/sharding has been created. 

**Test run log**
```
2024-03-15T10:08:35.464+0800	INFO	gowaku.relay	relay/waku_relay.go:247	gossipsub subscription	{"pubsubTopic": "/waku/2/go/pm/test"}
2024-03-15T10:08:35.466+0800	INFO	gowaku.peer-manager	peermanager/topic_event_handler.go:25	handleNewRelayTopicSubscription	{"pubSubTopic": "/waku/2/go/pm/test"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x40 pc=0x101377e58]

goroutine 2424 [running]:
github.com/waku-org/go-waku/waku/v2/protocol/metadata.(*WakuMetadata).RelayShard(0x5?)
	/Users/roman/sources/waku-org/go-waku/waku/v2/protocol/metadata/waku_metadata.go:82 +0x18
github.com/waku-org/go-waku/waku/v2/peermanager.(*PeerManager).handlerPeerTopicEvent(0x1400038fad0, {{0x1017c17b6, 0x12}, {0x14001d67560, 0x22}, 0x0})
	/Users/roman/sources/waku-org/go-waku/waku/v2/peermanager/topic_event_handler.go:124 +0x42c
github.com/waku-org/go-waku/waku/v2/peermanager.(*PeerManager).peerEventLoop(0x1400038fad0, {0x101d54bf8, 0x1400011e018})
	/Users/roman/sources/waku-org/go-waku/waku/v2/peermanager/topic_event_handler.go:174 +0x1e0
created by github.com/waku-org/go-waku/waku/v2/peermanager.TestHandlePeerTopicEvent
	/Users/roman/sources/waku-org/go-waku/waku/v2/peermanager/topic_event_handler_test.go:182 +0x5c8
exit status 2
FAIL	github.com/waku-org/go-waku/waku/v2/peermanager	8.491s
```

--- a/waku/v2/peermanager/topic_event_handler.go
+++ b/waku/v2/peermanager/topic_event_handler.go
@@ -121,24 +121,20 @@ func (pm *PeerManager) handlerPeerTopicEvent(peerEvt relay.EvtPeerTopic) {
 	wps := pm.host.Peerstore().(*wps.WakuPeerstoreImpl)
 	peerID := peerEvt.PeerID
 	if peerEvt.State == relay.PEER_JOINED {
-		rs, err := pm.metadata.RelayShard()
-		if err != nil {
-			pm.logger.Error("could not obtain the cluster and shards of wakunode", zap.Error(err))
-			return
-		} else if rs == nil {
-			pm.logger.Info("not using sharding")
-			return
-		}
-
-		if pm.metadata != nil && rs.ClusterID != 0 {
-			ctx, cancel := context.WithTimeout(pm.ctx, 7*time.Second)
-			defer cancel()
-			if err := pm.metadata.DisconnectPeerOnShardMismatch(ctx, peerEvt.PeerID); err != nil {
-				return
+		if pm.metadata != nil {
+			rs, err := pm.metadata.RelayShard()
+			if err != nil {
+				pm.logger.Error("could not obtain the cluster and shards of wakunode", zap.Error(err))
+			} else if rs != nil && rs.ClusterID != 0 {
+				ctx, cancel := context.WithTimeout(pm.ctx, 7*time.Second)
+				defer cancel()
+				if err := pm.metadata.DisconnectPeerOnShardMismatch(ctx, peerEvt.PeerID); err != nil {
+					return
+				}
 			}
 		}
 
-		err = wps.AddPubSubTopic(peerID, peerEvt.PubsubTopic)
+		err := wps.AddPubSubTopic(peerID, peerEvt.PubsubTopic)
 		if err != nil {
 			pm.logger.Error("failed to add pubSubTopic for peer",
 				logging.HostID("peerID", peerID), zap.String("topic", peerEvt.PubsubTopic), zap.Error(err))

